packages/filter/src/query.ts

```diff
@@ -560,16 +562,21 @@ export class FilterBuilder<MT extends object = AnyObject> {
    * @param f - A field name to be included, an array of field names to be
    * included, or an Fields object for the inclusion/exclusion
    */
-  fields(...f: (Fields<MT> | (keyof MT)[] | keyof MT)[]): this {
+  fields(...f: (Fields<MT> | Extract<keyof MT, string>)[]): this {
     if (!this.filter.fields) {
       this.filter.fields = {};
+    } else if (Array.isArray(this.filter.fields)) {
+      this.filter.fields = this.filter.fields.reduce(
+        (prev, current) => ({...prev, [current]: true}),
+        {},
```

Comment:
As I mentioned above, I think we should keep `fields` as an array, no reducing should be needed here.

Reply:
Good point 👍🏻 As I wrote in https://github.com/strongloop/loopback-next/pull/6517#discussion_r502372413, you convinced me it's better to store fields as an object internally.